In [6]:
import joblib
# Load the saved model & scaler
model = joblib.load("C:/Users/gudip/student_grade_model.pkl")
scaler = joblib.load("C:/Users/gudip/scaler.pkl")

In [9]:
import pandas as pd
from sklearn.metrics import mean_squared_error, r2_score

# importing the validation set
X = pd.read_csv('C:/Users/gudip/X Validation.csv')
y = pd.read_csv('C:/Users/gudip/Y Validation.csv')

In [11]:
# predicting the validation set
y_pred = model.predict(X)

C:\Users\gudip\anaconda3\Lib\site-packages\sklearn\utils\validation.py:2732: UserWarning: X has feature names, but RandomForestRegressor was fitted without feature names
  warnings.warn(


In [12]:
# calculating the metrics
mse = mean_squared_error(y, y_pred)
r2 = r2_score(y, y_pred)

# printing the metrics
print(f"Mean Squared Error: {mse}")
print(f"R-squared: {r2}")

Mean Squared Error: 644.2189490384616
R-squared: -7.583575791112732


Model performance metrics show that the **Mean Squared Error (MSE)** is extremely large at **644.22**, meaning that model predictions are quite far from true values on average. This is a very large error, meaning that the model may not be suitable for validation data or perhaps there are strong outliers in data.

The **R-squared statistic** is **-7.58**, which is negative, and indicates that the model does worse than a mean-based model would (which would have an R² of 0). This is a poor fit to the validation data, and the model is accounting for less variance than would a baseline model.

In short, these numbers mean that the model's predictive accuracy is extremely low on the validation set. It fails to detect the underlying trends well, and some more **tuning** or **feature engineering** may be necessary.


In [13]:
# Saving Data to Database
from sqlalchemy import create_engine

host = r'127.0.0.1' # denotes that the db in a local installation
db = r'MSDS610' # db we just created
user = r'postgres' # using the postgres user for this demo
pw = r'Ashok%403999' # this is the password established during installation
port = r'5432' # default port estabalished during install
schema = r'analytics' # schema we just created


# create a connection to the database
engine = create_engine(f'postgresql://{user}:{pw}@{host}:{port}/{db}')
db_conn = engine.connect()

In [14]:
X.to_sql('validation_predictions', engine, schema=schema, if_exists='replace', index=False)

# Verify the data was saved correctly
query = f"SELECT * FROM {schema}.validation_predictions LIMIT 5"
pd.read_sql(query, db_conn)

Socioeconomic Score  Study Hours  Sleep Hours  Attendance (%)  \
0              0.84372          4.1          8.7            58.0   
1              0.49081          4.5          8.6            47.0   
2              0.74071          6.9          6.5            69.0   
3              0.33208          1.0          7.9            57.0   
4              0.58595          9.3          6.8            59.0   

   Study_Attendance_Interaction  Sleep_Category_2  Sleep_Category_3  
0                         237.8             False              True  
1                         211.5             False              True  
2                         476.1              True             False  
3                          57.0             False              True  
4                         548.7              True             False